src/lib/xocl-main.c

```diff
@@ -125,11 +133,52 @@ static void xocl_drv_unregister_driver(enum xocl_subdev_id id)
 	pr_info("unregistered %s subdev driver\n", drvname);
 }
 
+int xocl_subdev_register_external_driver(enum xocl_subdev_id id, struct platform_driver *drv)
+{
+	int i;
+	int result = 0;
+
+	mutex_lock(&xocl_class_lock);
+	for (i = 0; i < ARRAY_SIZE(xocl_drv_maps); i++) {
+		struct xocl_drv_map *map = &xocl_drv_maps[i];
+
+		if (map->id != id)
+			continue;
+		if (map->drv) {
+			result = -EEXIST;
+			goto out;
+		}
+		map->drv = drv;
+	}
+out:
+	mutex_unlock(&xocl_class_lock);
+	return 0;
+}
+EXPORT_SYMBOL_GPL(xocl_subdev_register_external_driver);
+
+void xocl_subdev_unregister_external_driver(enum xocl_subdev_id id)
```

Comment:
also needs to call xocl_drv_unregister_driver() inside?

Reply:
Done